### PR TITLE
Allow for constrains with `validatedBy = MyValidator.class` to load validator from the bean context

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: validation/src/main/java/io/micronaut/validation/validator/ValidatorConfiguration.java [hunk 1]
```diff
@@ -49,6 +50,12 @@ public interface ValidatorConfiguration extends ConversionServiceProvider {
     @NonNull
     ConstraintValidatorRegistry getConstraintValidatorRegistry();
 
+    /**
+     * @return The constraint factory to use.
+     */
+    @NonNull
+    InternalConstraintValidatorFactory getConstraintValidatorFactory();
```
Comment (sdelamo) [inline validation/src/main/java/io/micronaut/validation/validator/ValidatorConfiguration.java:57]: It is weird we expose a method returning a @Internal Class in a public API?

**Review by sdelamo — COMMENTED** [review]
I creaed several minor refactors: 

https://github.com/micronaut-projects/micronaut-validation/pull/289
https://github.com/micronaut-projects/micronaut-validation/pull/290
https://github.com/micronaut-projects/micronaut-validation/pull/291

File: validation/src/main/java/io/micronaut/validation/validator/ValidatorConfiguration.java (hunk 1 above)
Comment (graemerocher) [reply validation/src/main/java/io/micronaut/validation/validator/ValidatorConfiguration.java:57]: can you annotate the method with `@Internal` and/or preferably make it non-public?

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: validation/src/main/java/io/micronaut/validation/validator/ValidatorConfiguration.java (hunk 1 above)
Comment (dstepanov) [reply validation/src/main/java/io/micronaut/validation/validator/ValidatorConfiguration.java:57]: I can expose ConstraintValidatorFactory and fail if its not InternalConstraintValidatorFactory

**Comment by dstepanov** [comment]
I changed it to be non-internal. Technically we can support API interface in the future

File: validation/src/main/java/io/micronaut/validation/validator/DefaultValidatorConfiguration.java [hunk 2]
```diff
@@ -133,6 +139,24 @@ public ConstraintValidatorRegistry getConstraintValidatorRegistry() {
         return constraintValidatorRegistry;
     }
 
+    @Override
+    public InternalConstraintValidatorFactory getConstraintValidatorFactory() {
```
Comment (sdelamo) [inline validation/src/main/java/io/micronaut/validation/validator/DefaultValidatorConfiguration.java:143]: should not we change the return type to be `ConstraintValidationFactory`?

File: validation/src/main/java/io/micronaut/validation/validator/ValidatorConfiguration.java [hunk 3]
```diff
@@ -49,6 +50,12 @@ public interface ValidatorConfiguration extends ConversionServiceProvider {
     @NonNull
     ConstraintValidatorRegistry getConstraintValidatorRegistry();
 
+    /**
+     * @return The constraint factory to use.
+     */
```
Comment (sdelamo) [inline validation/src/main/java/io/micronaut/validation/validator/ValidatorConfiguration.java:55]: Add `@since`

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**